consensus/src/simplex/mod.rs

```diff
@@ -205,6 +206,81 @@ mod tests {
     };
     use tracing::debug;
 
+    async fn register_validators(
+        oracle: &mut Oracle,
+        validators: &[Bytes],
+    ) -> HashMap<Bytes, ((Sender, Receiver), (Sender, Receiver))> {
+        let mut registrations = HashMap::new();
+        for validator in validators.iter() {
+            let (voter_sender, voter_receiver) =
+                oracle.register(validator.clone(), 0).await.unwrap();
+            let (resolver_sender, resolver_receiver) =
+                oracle.register(validator.clone(), 1).await.unwrap();
+            registrations.insert(
+                validator.clone(),
+                (
+                    (voter_sender, voter_receiver),
+                    (resolver_sender, resolver_receiver),
+                ),
+            );
+        }
+        registrations
+    }
+
+    enum Action {
+        NewLink(Link),
+        Relink(Link),
+        Unlink,
+    }
+
+    enum Scope {
```

Comment:
Can't attest it would be "cleaner" but give we don't do anything with `All`, it may be simpler to just do `Option<fn(usize, usize) -> bool>`?

Reply:
yeah makes sense since before I had more options here like `Only` and `Omit` that made passing-in certain types of functions easier